### PR TITLE
#2281089 Added entity_is_new condition and tests.

### PR DESCRIPTION
**Pull request by dermario**
This is my implementation of https://www.drupal.org/node/2281089

### CODE REVIEW

File: src/Plugin/Condition/EntityIsNew.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * @Condition(
+ *   id = "rules_entity_is_new",
+ *   label = @Translation("Entity is new"),
+ *   category = @Translation("Entity"),
+ *   context = {
+ *     "entity" = @ContextDefinition("entity",
+ *       label = @Translation("Entity"),
+ *       description = @Translation("Specifies the entity for which to evaluate the condition.")
+ *     ),
+ *     "is_new" = @ContextDefinition("boolean",
```
Comment (klausi) [inline src/Plugin/Condition/EntityIsNew.php]: That context is not required. See also the D7 implementation which also has only one parameter. If people want the condition to be reversed they can just negate it.

File: tests/src/Integration/Condition/EntityIsNewTest.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+   * @var \Drupal\rules\Engine\RulesConditionInterface
+   */
+  protected $condition;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
```
Comment (fago) [inline tests/src/Integration/Condition/EntityIsNewTest.php]: Unnecessary empty line.

File: tests/src/Integration/Condition/EntityIsNewTest.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+   * @covers ::evaluate()
+   */
+  public function testConditionEvaluation() {
+    $entity = $this->getMock('Drupal\Core\Entity\EntityInterface');
+    $entity->expects($this->once())
+      ->method('isNew')
+      ->will($this->returnValue(TRUE));
+
+    // Add the test node to our context as the evaluated entity.
+    $this->condition->setContextValue('entity', $this->getMockTypedData($entity));
```
Comment (fago) [inline tests/src/Integration/Condition/EntityIsNewTest.php]: That's bad practice (and needs to be removed from more tests).
setContextValue() is fine with $entity, nothing else required.

File: src/Plugin/Condition/EntityIsNew.php [hunk 4]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Condition\EntityIsOfType.
```
Comment (fubhy) [inline src/Plugin/Condition/EntityIsNew.php]: Should be EntityIsNew

File: tests/src/Integration/Condition/EntityIsNewTest.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+namespace Drupal\Tests\rules\Integration\Condition;
+
+use Drupal\Tests\rules\Integration\RulesIntegrationTestBase;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Condition\EntityIsNew
+ * @group rules_conditions
+ */
+class EntityIsNewTest extends RulesIntegrationTestBase {
```
Comment (fubhy) [inline tests/src/Integration/Condition/EntityIsNewTest.php]: Should extend RulesEntityIntegrationTestBase

**Comment by fubhy** [comment]
Thanks. I am going to merge this once you squash the commits as discussed on IRC earlier. Let me know if you need assistance in doing so.

**Pull request closed without merging**